packages/serverpod/lib/src/database/database_connection.dart

```diff
@@ -295,26 +295,23 @@ class DatabaseConnection {
   }) async {
     var startTime = DateTime.now();
 
-    var data = row.toJsonForDatabase();
+    var selectedColumns = columns ?? row.table.columns;
+    Map data = row.allToJson();
 
-    columns?.forEach((column) {
-      if (!data.containsKey(column.columnName)) {
-        throw Exception(
-          'Column ${column.columnName} does not exist in table row provided.',
-        );
-      }
-    });
-
-    int? id = data['id'];
-
-    Iterable<String> cols = columns?.map((c) => c.columnName) ?? data.keys;
+    for (var column in selectedColumns) {
+      assert(data.containsKey(column.columnName));
+    }
 
-    String updates = cols.where((column) => column != 'id').map((column) {
-      var value = DatabasePoolManager.encoder.convert(data[column]);
-      return '"$column" = $value';
+    var updates = selectedColumns
+        .where((column) => column.columnName != 'id')
+        .map((column) {
+      var value = DatabasePoolManager.encoder.convert(data[column.columnName]);
+      return '"${column.columnName}" = $value';
```

Comment:
Suggestion: Just use `'$column'` here as name to use the column formatting name.

Reply:
Same comment applies for insert.

Reply:
I had some issues getting that to work, do I have to give the query a name the same name as is included in the column for this to work?

Reply:
Hmm, should just be to use it as a string, look at this line where we create an alias for it in the find call: https://github.com/serverpod/serverpod/blob/7665f3da10e2e5e5fa248fc5158bd1385c5b7ac6/packages/serverpod/lib/src/database/database_query.dart#L337

Reply:
not valid in postgresql!